File: clients/python/model/unit.py
from .vec2_double import Vec2Double
from .vec2_double import Vec2Double
from .jump_state import JumpState
from .weapon import Weapon
class Unit:
    def __init__(self, player_id, id, health, position, size, jump_state, walked_right, stand, on_ground, on_ladder, mines, weapon):
        self.player_id = player_id
        self.id = id
        self.health = health
        self.position = position
        self.size = size
        self.jump_state = jump_state
        self.walked_right = walked_right
        self.stand = stand
        self.on_ground = on_ground
        self.on_ladder = on_ladder
        self.mines = mines
        self.weapon = weapon
    @staticmethod
    def read_from(stream):
        player_id = stream.read_int()
        id = stream.read_int()
        health = stream.read_int()
        position = Vec2Double.read_from(stream)
        size = Vec2Double.read_from(stream)
        jump_state = JumpState.read_from(stream)
        walked_right = stream.read_bool()
        stand = stream.read_bool()
        on_ground = stream.read_bool()
        on_ladder = stream.read_bool()
        mines = stream.read_int()
        if stream.read_bool():
            weapon = Weapon.read_from(stream)
        else:
            weapon = None
        return Unit(player_id, id, health, position, size, jump_state, walked_right, stand, on_ground, on_ladder, mines, weapon)
    def write_to(self, stream):
        stream.write_int(self.player_id)
        stream.write_int(self.id)
        stream.write_int(self.health)
        self.position.write_to(stream)
        self.size.write_to(stream)
        self.jump_state.write_to(stream)
        stream.write_bool(self.walked_right)
        stream.write_bool(self.stand)
        stream.write_bool(self.on_ground)
        stream.write_bool(self.on_ladder)
        stream.write_int(self.mines)
        if self.weapon is None:
            stream.write_bool(False)
        else:
            stream.write_bool(True)
            self.weapon.write_to(stream)
    def __repr__(self):
        return "Unit(" + \
            repr(self.player_id) + "," + \
            repr(self.id) + "," + \
            repr(self.health) + "," + \
            repr(self.position) + "," + \
            repr(self.size) + "," + \
            repr(self.jump_state) + "," + \
            repr(self.walked_right) + "," + \
            repr(self.stand) + "," + \
            repr(self.on_ground) + "," + \
            repr(self.on_ladder) + "," + \
            repr(self.mines) + "," + \
            repr(self.weapon) + \
            ")"

File: clients/javascript/model/unit.js
const Vec2Double = require('./vec2-double').Vec2Double;
const JumpState = require('./jump-state').JumpState;
const Weapon = require('./weapon').Weapon;
class Unit {
    constructor(playerId, id, health, position, size, jumpState, walkedRight, stand, onGround, onLadder, mines, weapon) {
        this.playerId = playerId;
        this.id = id;
        this.health = health;
        this.position = position;
        this.size = size;
        this.jumpState = jumpState;
        this.walkedRight = walkedRight;
        this.stand = stand;
        this.onGround = onGround;
        this.onLadder = onLadder;
        this.mines = mines;
        this.weapon = weapon;
    }
    static async readFrom(stream) {
        let playerId;
        playerId = await stream.readInt();
        let id;
        id = await stream.readInt();
        let health;
        health = await stream.readInt();
        let position;
        position = await Vec2Double.readFrom(stream);
        let size;
        size = await Vec2Double.readFrom(stream);
        let jumpState;
        jumpState = await JumpState.readFrom(stream);
        let walkedRight;
        walkedRight = await stream.readBool();
        let stand;
        stand = await stream.readBool();
        let onGround;
        onGround = await stream.readBool();
        let onLadder;
        onLadder = await stream.readBool();
        let mines;
        mines = await stream.readInt();
        let weapon;
        if (await stream.readBool()) {
            weapon = await Weapon.readFrom(stream);
        } else {
            weapon = null;
        }
        return new Unit(playerId, id, health, position, size, jumpState, walkedRight, stand, onGround, onLadder, mines, weapon);
    }
    async writeTo(stream) {
        let playerId = this.playerId;
        await stream.writeInt(playerId);
        let id = this.id;
        await stream.writeInt(id);
        let health = this.health;
        await stream.writeInt(health);
        let position = this.position;
        await position.writeTo(stream);
        let size = this.size;
        await size.writeTo(stream);
        let jumpState = this.jumpState;
        await jumpState.writeTo(stream);
        let walkedRight = this.walkedRight;
        await stream.writeBool(walkedRight);
        let stand = this.stand;
        await stream.writeBool(stand);
        let onGround = this.onGround;
        await stream.writeBool(onGround);
        let onLadder = this.onLadder;
        await stream.writeBool(onLadder);
        let mines = this.mines;
        await stream.writeInt(mines);
        let weapon = this.weapon;
        if (weapon === null) {
            await stream.writeBool(false);
        } else {
            await stream.writeBool(true);
            await weapon.writeTo(stream);
        }
    }
}
module.exports = { Unit: Unit }
